### PR TITLE
Monster global constants cleanup

### PR DESCRIPTION
**Pull request by aetn23**
Changed defines in files monster.cpp and monster.h to be constinit const int instead. Fixed some typos along the way. Changed passing the reference to the first element of an array to simply passing the array itself or its underlying data in the InitMonsterGFX.

Removing defines is good practice in modern C++. Other changes simplify code, or fix language errors in the comments.

### CODE REVIEW

File: Source/monster.h [hunk 1]
```diff
@@ -25,8 +25,8 @@ namespace devilution {
 
 struct Missile;
 
-#define MAXMONSTERS 200
-#define MAX_LVLMTYPES 24
+constinit const int MaxMonsters = 200;
+constinit const int maxLvlmtypes = 24;
```
Comment (qndel) [inline Source/monster.h:29]: why lowercase

**Comment by AJenbo** [comment]
`constinit` is C++20, we currently have a requirement slightly below C++17.

**Review by AJenbo — APPROVED** [review]
Nice :+1:

File: Source/monster.h [hunk 2]
```diff
@@ -25,8 +25,8 @@ namespace devilution {
 
 struct Missile;
 
-#define MAXMONSTERS 200
-#define MAX_LVLMTYPES 24
+constexpr const int MaxMonsters = 200;
+constexpr const int MaxLvlMTypes = 24;
```
Comment (ephphatha) [inline Source/monster.h:29]: constexpr implies const, no need to repeat in these definitions.

Comment (AJenbo) [reply Source/monster.h:29]: @aetn23 would you be willing to do a follow up to correct this :)

Comment (aetn23) [reply Source/monster.h:29]: Sure

**Pull request merged**